torchgeo/models/dofa.py

```diff
@@ -0,0 +1,544 @@
+
+    Reference implementation:
+
+    * https://github.com/ShadowXZT/DOFA-Net
+
+    If you use this model in your research, please cite the following paper:
+
+    * TODO
+
+    .. versionadded:: 0.6
+    """
+
+    def __init__(
+        self,
+        img_size: int = 224,
+        patch_size: int = 16,
+        drop_rate: float = 0.0,
+        embed_dim: int = 1024,
+        depth: int = 24,
+        num_heads: int = 16,
+        wv_planes: int = 128,
+        num_classes: int = 45,
+        global_pool: bool = True,
+        mlp_ratio: float = 4.0,
+        norm_layer: type[nn.Module] = nn.LayerNorm,
+    ) -> None:
+        """Initialize a new OFAViT instance.
+
+        Args:
+            img_size: Input image size.
+            patch_size: Patch size.
+            drop_rate: Head dropout rate.
+            embed_dim: Transformer embedding dimension.
+            depth: Depth of transformer.
+            num_heads: Number of attention heads.
+            wv_planes: Number of wavelengths.
+            num_classes: Number of classes for classification head.
+            global_pool: Whether or not to perform global pooling.
+            mlp_ratio: Ratio of MLP hidden dim to embedding dim.
+            norm_layer: Normalization layer.
+        """
+        super().__init__()
+
+        self.wv_planes = wv_planes
+        self.global_pool = global_pool
+        if self.global_pool:
+            norm_layer = norm_layer
+            embed_dim = embed_dim
+            self.fc_norm = norm_layer(embed_dim)
+        else:
+            self.norm = norm_layer(embed_dim)
+
+        # --------------------------------------------------------------------------
+        # MAE encoder specifics
+        self.patch_embed = Dynamic_MLP_OFA(
+            wv_planes=128, inter_dim=128, kernel_size=16, embed_dim=embed_dim
+        )
+        self.num_patches = (img_size // patch_size) ** 2
+        self.cls_token = nn.Parameter(torch.zeros(1, 1, embed_dim))
+        # ---------------------------------------------------------------------------
+        self.pos_embed = nn.Parameter(
+            torch.zeros(1, self.num_patches + 1, embed_dim), requires_grad=False
+        )  # fixed sin-cos embedding
+
+        self.blocks = nn.ModuleList(
+            [
+                Block(
+                    embed_dim,
+                    num_heads,
+                    mlp_ratio,
+                    qkv_bias=True,
+                    norm_layer=norm_layer,
+                )
+                for i in range(depth)
+            ]
+        )
+
+        self.head_drop = nn.Dropout(drop_rate)
+        self.head = (
+            nn.Linear(embed_dim, num_classes) if num_classes > 0 else nn.Identity()
+        )
+
+    def forward_features(self, x: Tensor, wave_list: list[float]) -> Tensor:
+        """Forward pass of the feature embedding layer.
+
+        Args:
+            x: Input mini-batch.
+            wave_list: List of wavelengths.
```

Comment:
Can we rename this to `wavelengths`? I think it's more clear. The type hints already make it clear that it's a list.

Reply:
ok